pkg/band/rpc.go

```diff
@@ -58,3 +59,138 @@ func GetPhyVersions(ctx context.Context, req *ttnpb.GetPhyVersionsRequest) (*ttn
 	}
 	return res, nil
 }
+
+// ListBands returns the list of supported bands.
+func ListBands(ctx context.Context, req *ttnpb.ListBandsRequest) (*ttnpb.ListBandsResponse, error) {
+	filteredVersions := make(map[string]map[ttnpb.PHYVersion]Band)
+
+	if req.BandId != "" {
+		versions, ok := All[req.BandId]
+		if !ok {
+			return nil, errBandNotFound.WithAttributes(req.BandId, req.PhyVersion.String())
+		}
+
+		filteredVersions[req.BandId] = versions
+	}
+
+	if len(filteredVersions) == 0 {
+		filteredVersions = All
+	}
+
+	if req.PhyVersion != 0 {
+		for bandID, versions := range filteredVersions {
+			version, ok := versions[req.PhyVersion]
+			if !ok {
+				delete(filteredVersions, bandID)
+				continue
+			}
+			filteredVersions[bandID] = make(map[ttnpb.PHYVersion]Band)
+			filteredVersions[bandID][req.PhyVersion] = version
+		}
+	}
+
+	res := &ttnpb.ListBandsResponse{
+		Descriptions: make(map[string]*ttnpb.ListBandsResponse_VersionedBandDescription),
+	}
+
+	for bandID, versions := range filteredVersions {
+		versionedBandDescription := &ttnpb.ListBandsResponse_VersionedBandDescription{
+			Band: make(map[string]*ttnpb.BandDescription),
+		}
+
+		for PHYVersion, band := range versions {
+			versionedBandDescription.Band[PHYVersion.String()] = band.convertToBandDescription()
+		}
+
+		res.Descriptions[bandID] = versionedBandDescription
+	}
+
+	return res, nil
+}
+
+// convertToBandDescription parses a band into a ttnpb.BandDescription
+func (band Band) convertToBandDescription() *ttnpb.BandDescription {
+	bandDescription := &ttnpb.BandDescription{
+		Id: band.ID,
+		Beacon: &ttnpb.BandDescription_Beacon{
+			DataRateIndex:    band.Beacon.DataRateIndex,
+			CodingRate:       band.Beacon.CodingRate,
+			InvertedPolarity: band.Beacon.InvertedPolarity,
+		},
+		PingSlotFrequency:      &types.UInt64Value{},
```

Comment:
Leave this `nil` here and initialize the full value below. Otherwise you will report a `0`/`false` when the real value is `null` / nonexistant.